addons/mail/static/src/js/systray/systray_activity_menu.js

```diff
@@ -111,13 +111,18 @@ var ActivityMenu = Widget.extend({
         if (actionXmlid) {
             this.do_action(actionXmlid);
         } else {
+            var domain = [['activity_ids.user_id', '=', session.uid]]
+            if (targetAction.data('domain') != null) {
+                domain.unshift('&', targetAction.data('domain'))
```

Comment:
data('domain') should be a valid domain, not a leaf. Indeed I see you created a lead of form ['activity_ids.res_id', 'in', ids] whereas it should support a complete domain like [['activity_ids.res_id', 'in', ids]] (list of tuples / list of list).

Xmo thinks there are domain tools available for JS, but neither him or me found them. Maybe you could ask to JS team how to combine two domain parts in a clean way, as you can't be sure data('domain') will be well formed.

Reply:
Updated data('domain') to be a proper domain. There were 2 techniques the JS team suggested to me for combining 2 domains:

`domain1 = domain1.concat(domain2)` or `domain1 = [...domain1, ...domain2].`

I went with the first one.